fms_extras/utils/generation.py

```diff
@@ -125,43 +130,45 @@ def speculative_generate(
         # create candidate sequences
         child_sequence_ids_list = []
         child_sequence_ids_flattened = []
-        num_tokens_per_sequence = [n_adds for _ in range(input_ids.size(0) * top_k)]
-        # each parent will have top_k child sequences
+        num_tokens_per_sequence = [inp_len for _ in range(input_ids.size(0) * n_candidates)]
+        # each parent will have n_candidates child sequences
         for parent_sequence_id in parent_sequence_ids:
             child_sequence_ids = kv_cache_manager.add_child_sequences(
-                parent_sequence_id, top_k
+                parent_sequence_id, n_candidates
             )
             child_sequence_ids_list.append(child_sequence_ids)
             child_sequence_ids_flattened.extend(child_sequence_ids)
 
-        # add n_adds tokens to each candidate
+        # add inp_len tokens to each candidate
         cache_data = kv_cache_manager.allocate_tokens(
             num_tokens_per_sequence, child_sequence_ids_flattened
         )
         position_ids = cache_data.compute_position_ids(num_tokens_per_sequence)
 
         # Get candidate set of speculations
-        adds = speculator.generate_suffixes(embeds, input_ids, threshes, top_k)  # b k h
+        input_ids = speculator.generate_suffixes(embeds, input_ids, threshes, n_candidates)  # b k h
```

Comment:
I see you are setting the `generate_suffixes` to input_ids then calling cat with the same input_ids as before, overwriting here will not cause issues on the next cat? Does ths assume adds will always be the same as the original input_ids?

Reply:
Previously, we created `adds` and then immediately concatenated to `input_ids` and assigned that to `input_ids`. Then `adds` never got used again. So this is cleaner, and `input_ids` was getting overwritten anyways

Reply:
oh nvm I see what you're saying